src/jdk.compiler/share/classes/com/sun/tools/javac/parser/JavacParser.java

```diff
@@ -234,25 +235,41 @@ protected DocCommentTable newDocCommentTable(boolean keepDocComments, ParserFact
     JCVariableDecl receiverParam;
 
     /** When terms are parsed, the mode determines which is expected:
-     *     mode = EXPR        : an expression
-     *     mode = TYPE        : a type
-     *     mode = NOPARAMS    : no parameters allowed for type
-     *     mode = TYPEARG     : type argument
-     *     mode |= NOLAMBDA   : lambdas are not allowed
+     *     mode = EXPR          : an expression
+     *     mode = TYPE          : a type
+     *     mode = NOPARAMS      : no parameters allowed for type
+     *     mode = TYPEARG       : type argument
+     *     mode |= NOLAMBDA     : lambdas are not allowed
      */
-    protected static final int EXPR = 0x1;
-    protected static final int TYPE = 0x2;
-    protected static final int NOPARAMS = 0x4;
-    protected static final int TYPEARG = 0x8;
-    protected static final int DIAMOND = 0x10;
-    protected static final int NOLAMBDA = 0x20;
+    protected static final int EXPR          = 1 << 0;
```

Comment:
I note how many of the changes in this class are "stylistic" - e.g. replacing direct manipulation of state bits with method calls. While I'm not opposed to it, this is something rather orthogonal to what's being discussed here (perhaps we should consider factoring it out?)

Reply:
I agree. I haven't updated uses by others, so it would be good to coordinate.